### PR TITLE
revive the book for bookdown::render_book

### PR DESCRIPTION
**Pull request by dongzhuoer**
Add some modifications so that the book can be built by `bookdown::render_book()`

### CODE REVIEW

**Comment by hadley** [comment]
It's been a while since I looked at this repo, but would you still be interested in finishing this off?

**Comment by dongzhuoer** [comment]
Yeah, I feel quiet honored. Since it quite a good resource for learning ggplot2, and publish it by bookdown can permit more people open access to it.

File: data-manip.Rmd [hunk 1]
```diff
@@ -1,10 +1,9 @@
 ---
```
Comment (hadley) [inline data-manip.Rmd]: Can you please remove the yaml header from all the pages?

File: data-manip.Rmd [hunk 2]
```diff
@@ -1,10 +1,9 @@
...
 title: Data transformation
-output: oldbookdown::html_chapter
-bibliography: references.bib
+knit: "bookdown::preview_chapter"
 ---
 
-```{r data, include = FALSE}
+```{r data-manip, include = FALSE}
 chapter <- "data-manip"
```
Comment (hadley) [inline data-manip.Rmd]: I think we should also stop using `chapter`, and remove the custom `fig.path` and `cache.path` in `common.R`.

File: index.Rmd [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
```
Comment (hadley) [inline index.Rmd]: Can you please put the `output` related metadata in `_output.yml`? I think this file should containing something more like this:

```yaml
knit: "bookdown::render_book"
title: "ggplot2: elegant graphics for data analysis"
author: ["Hadley Wickham"]
url: 'https\://r4ds.had.co.nz/'
github-repo: hadley/ggplot2-book
twitter-handle: hadley
site: bookdown::bookdown_site
documentclass: book
```

File: _bookdown.yml [hunk 4]
```diff
@@ -0,0 +1 @@
+book_filename: ggplot2
```
Comment (hadley) [inline _bookdown.yml]: I don't normally bother with customising the file name, but this file should contain `new_session: yes`, and should list the `rmd_files` in order.

**Comment by hadley** [comment]
Superseeded by #144

**Pull request closed without merging**